src/pages/Signup.jsx

```diff
@@ -0,0 +1,304 @@
+  });
+  return response.json();
+};
+
+const initializePod = async (webId, podUrl, caseManagerWebId, fetch) => {
+  try {
+    await createContainerAt(`${podUrl}PASS`, { fetch });
+  } finally {
+    const datasetWithAcl = await getSolidDatasetWithAcl(`${podUrl}PASS/`, { fetch });
+    let acl = getResourceAcl(datasetWithAcl) ?? createAcl(datasetWithAcl);
+
+    acl = setAgentResourceAccess(acl, webId, {
+      read: true,
+      append: true,
+      write: true,
+      control: true
+    });
+    acl = setAgentDefaultAccess(acl, webId, {
+      read: true,
+      append: true,
+      write: true,
+      control: true
+    });
+
+    if (caseManagerWebId) {
+      acl = setAgentResourceAccess(acl, caseManagerWebId, {
+        read: true,
+        append: true,
+        write: true,
+        control: false
+      });
+      acl = setAgentDefaultAccess(acl, caseManagerWebId, {
+        read: true,
+        append: true,
+        write: true,
+        control: false
+      });
+    }
+    await saveAclFor(datasetWithAcl, acl, { fetch });
+  }
+};
+
+const ShowNewPod = ({ oidcIssuer }) => {
+  const { session } = useSession();
+  const { data, isSuccess } = useContactsList();
+
+  return (
+    <Typography>
+      <h1>You have successfully registered for a pod.</h1>
+      <br />
+      You can find your pod here: {oidcIssuer}
+      <br />
+      Your webId is: {session.info.webId}
+      <br />
+      {isSuccess && data.length > 0
+        ? `You have registered with ${data[0].person}`
+        : 'You have not registered with a case manager'}
+    </Typography>
+  );
+};
+
+const PodRegistrationForm = ({ register, caseManagerName }) => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+
+  const [searchParams] = useSearchParams();
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    register(email, password, confirmPassword);
+  };
+
+  return (
+    <>
+      <Typography
+        display="flex"
+        justifyContent="center"
+        alignItems="center"
+        mb={2}
+        variant="h5"
+        component="h3"
+      >
+        Register For PASS
+      </Typography>
+      {searchParams.get('webId') ? (
+        <p>You will register with {caseManagerName ?? searchParams.get('webId')}</p>
+      ) : (
+        <p>You are not registering with anyone</p>
+      )}
+
+      <Card variant="outlined" sx={cardStyle}>
+        <CardHeader title="Set Up a New Pod" />
+
+        <form onSubmit={handleSubmit} style={formRowStyle} autoComplete="off">
+          <TextField
+            style={textFieldStyle}
+            id="email-form"
+            aria-label="Email"
+            label="Email"
+            variant="outlined"
+            value={email}
+            onChange={(e) => setEmail(e.target.value)}
+            required
+          />
+          <br />
+          <TextField
+            style={textFieldStyle}
```

Comment:
I think it would help security if we automatically hide the password as the user enters it. That seems pretty standard practice. And then it could include a button to reveal it. Like this:

![2023-08-21 (20)](https://github.com/codeforpdx/PASS/assets/83156697/8f34b516-73a8-439b-84dc-510c528da42f)

This also determines how strong the password is. We could add that as well. Probably not needed for MVP but might be nice to include down the line.

Another aspect to consider is whether we want to set minimum requirements for passwords, such as no fewer than 8 characters, at least one number, etc.

Reply:
I'll add this in. Is there a particular component you had in mind for handling these passwords?

Reply:
This should cover everything we need:
https://mui.com/material-ui/react-text-field/#input-adornments

Reply:
Sounds good. That won't handle the password evaluation at the bottom, but that can be a separate PR.

Reply:
Fixed